app/services/organization_update_service.rb

```diff
@@ -12,6 +12,10 @@ class << self
     def update(organization, params)
       return false unless valid?(organization, params)
 
+      if params.has_key?("partner_form_fields")
+        params["partner_form_fields"].delete_if { |field| field == "" }
+      end
+
```

Comment:
Hmm... no, I'd say this is the right place for it since it has to do with input munging (getting weird data from the frontend and fixing it) rather than field manipulation. Theoretically it could even be fixed in the controller, but I think the service class is a better place.